src/ansi-c/c_typecheck_base.cpp

```diff
@@ -395,10 +393,23 @@ void c_typecheck_baset::typecheck_redefinition_non_type(
       }
 
       if(new_symbol.is_macro)
+      {
         old_symbol.is_macro=true;
+      }
       else
+      {
+        // old symbol had no body but might have had a contract
+        if(to_code_with_contract_type(old_symbol.type).has_contract())
+        {
+          error().source_location = new_symbol.location;
+          error()
+            << "function '" << new_symbol.display_name()
+            << "' cannot have a declaration with a contract and a definition."
```

Comment:
I'm struggling to understand what this means: is it not permitted to have a function contract for a function that also has a definition? Or is it just that the definition must not have a contract attached to it? (But then I don't see where this is being checked.)